racket/collects/racket/match/struct.rkt

```diff
@@ -17,57 +34,43 @@
               [v (if (identifier? #'struct-name)
                      (syntax-local-value #'struct-name fail)
                      (fail))]
-              [field-acc->pattern (make-free-id-table)])
+              [field->pattern (make-hash)])
```

Comment:
Field names are distinct in each level of a struct. Since the `struct-field-info` protocol returns a list of symbols, it makes more sense to use hash keyed by symbols.